File: CarlaErik/UvA.SoftCon.Questionnaire/AST/Model/Expressions/Unary/UnaryExpression.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using UvA.SoftCon.Questionnaire.AST.Model.Statements;

namespace UvA.SoftCon.Questionnaire.AST.Model.Expressions.Unary
{
    public abstract class UnaryExpression : Node, IExpression
    {
        public Operation Operation
        {
            get;
            private set;
        }

        public IExpression Operand
        {
            get;
            private set;
        }

        public IExpression Right
        {
            get;
            private set;
        }

        public UnaryExpression(Operation operation, IExpression operand, TextPosition position)
            : base(position)
        {
            Operation = operation;
            Operand = operand;
        }

        public abstract DataType GetType(IDictionary<string, DataType> symbolTable);

        public abstract bool OperandTypeIsValid(DataType operandType);
    }
}

File: nlamah/src/main/java/org/nlamah/QL/Unary/UnaryExpression.java
package org.nlamah.QL.Unary;

import org.nlamah.QL.Expression.Expression;

public abstract class UnaryExpression extends Expression 
{
	private Expression expression;
	
	public UnaryExpression (Expression expression)
	{
		this.expression = expression;
	}
	
	public Expression expression()
	{
		return this.expression;
	}
}
